recipes-kernel/linux/4.19/patches/skb-forward-copy-bridge-param.patch

```diff
@@ -0,0 +1,84 @@
+################################################################################
+SHORT DESCRIPTION: 
+################################################################################
+Add a module parameter to enable skb forward copy.
+
+################################################################################
+LONG DESCRIPTION: 
+################################################################################
+When the parameter is enabled, the bridge will pass copy of the skb and release
+the initial one.
+
+################################################################################
+CHANGELOG 
+################################################################################
+Original author: Ross Phillipson <philipsonr@ainfosec.com>
+Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
+Port to 4.14: Richard Turner <turnerr@ainfosec.com>
+Port to 4.19: Richard Turner <turnerr@ainfosec.com>
+
+################################################################################
+REMOVAL 
+################################################################################
+?
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+None.
+
+################################################################################
+INTERNAL DEPENDENCIES 
+################################################################################
+None.
+
+################################################################################
+PATCHES 
+################################################################################
+--- a/net/bridge/br_forward.c
++++ b/net/bridge/br_forward.c
+@@ -11,6 +11,7 @@
+  *	2 of the License, or (at your option) any later version.
+  */
+ 
++#include <linux/module.h>
+ #include <linux/err.h>
+ #include <linux/slab.h>
+ #include <linux/kernel.h>
+@@ -21,6 +22,10 @@
+ #include <linux/netfilter_bridge.h>
+ #include "br_private.h"
+ 
++static int br_skb_forward_copy = 0;
++module_param_named(skb_forward_copy, br_skb_forward_copy, uint, S_IRUGO);
++MODULE_PARM_DESC(skb_forward_copy, "Enable or disable SKB copying on forward path");
++
+ /* Don't forward packets to originating port or forwarding disabled */
+ static inline int should_deliver(const struct net_bridge_port *p,
+ 				 const struct sk_buff *skb)
+@@ -37,6 +42,8 @@ static inline int should_deliver(const s
+ int br_dev_queue_push_xmit(struct net *net, struct sock *sk, struct sk_buff *skb)
+ {
+ 	skb_push(skb, ETH_HLEN);
++	struct sk_buff *new_skb;
```

Comment:
This declaration should be added before skb_push

Reply:
Agreed, fix coming.